dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/datavalueset/DefaultDataValueSetService.java

```diff
@@ -762,7 +763,7 @@ private ImportSummary saveDataValueSet( ImportOptions importOptions, JobConfigur
                     continue;
                 }
         
-                if ( dataSet != null && !aclService.canDataWrite( currentUser, dataSet ) )
+                if ( dataSet != null && !aclService.canDataRead( currentUser, dataSet ) )
```

Comment:
Hi @Melaeke ,

Why are we changing this?

Reply:
When data is exported from facility users and then being imported by users from Higher hierarchy, the users which are importing the data only have view access to the DS and not write access. I have changed canDataWrite to canDataRead, so as to support all users to import completness without granting them to write access to the dataSet which we don't want to give.

Reply:
then that is totally wrong.

Reply:
I will merge it for now. otherwise, we need to change the workflow. 

Any user supposed to write data (whether through data entry or import/export) need to have data_write access.